presto-main/src/main/java/io/prestosql/server/security/UserMapping.java

```diff
@@ -84,33 +86,57 @@ public UserMappingRules(
         }
     }
 
+    enum Case
+    {
+        KEEP(Function.identity()),
+        LOWER(value -> value.toLowerCase(Locale.ENGLISH)),
+        UPPER(value -> value.toUpperCase(Locale.ENGLISH)),
+        /**/;
+
+        private final Function<String, String> transformation;
+
+        Case(Function<String, String> transformation)
+        {
+            this.transformation = requireNonNull(transformation, "transformation is null");
+        }
+
+        public final String transform(String value)
+        {
+            return transformation.apply(value);
+        }
+    }
+
     public static final class Rule
     {
         private final Pattern pattern;
         private final String user;
         private final boolean allow;
+        private final Case kase;
```

Comment:
Alternatively, `capitalization`

Reply:
Or `userCase`, since this applies to the username.

Reply:
done